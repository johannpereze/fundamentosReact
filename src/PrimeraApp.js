import React from "react";
import PropTypes from "prop-types";

const PrimeraApp = ({ saludo, subtitulo = "Soy un subtitulo" }) => {
  return (
    <>
      <h1>{saludo}</h1>
      <p>M{subtitulo}</p>
    </>
  );
};
PrimeraApp.propTypes = {
  saludo: PropTypes.string.isRequired,
};

export default PrimeraApp;
