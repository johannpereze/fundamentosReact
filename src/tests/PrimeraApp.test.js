// import { render } from "@testing-library/react"
import PrimeraApp from "../PrimeraApp";
import { shallow } from "enzyme";

describe("Pruebas en PrimeraApp", () => {
  //test sin enzyme
  // test('Debe mostrar el mensaje "Hola Mundo"', () => {
  //     const saludo = 'Hola Mundo'
  //     const {getByText} = render(<PrimeraApp saludo={saludo} />)
  //     wrapper.getByText()
  //     expect(getByText(saludo)).toBeInTheDocument()
  // })

  //test con enzyme:
  test("Debe de mostrar PrimeraApp correctamente", () => {
    const saludo = "Hola Mundo";
    const wrapper = shallow(<PrimeraApp saludo={saludo} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("debe mostrar el subtitulo enviado por props", () => {
    const saludo = "Hola Mundo";
    const subtitulo = "Hola Mundo";

    const wrapper = shallow(
      <PrimeraApp saludo={saludo} subtitulo={subtitulo} />
    );
    const textoParrafo = wrapper.find("p").text();
    expect(textoParrafo).toBe(subtitulo);
  });
});
