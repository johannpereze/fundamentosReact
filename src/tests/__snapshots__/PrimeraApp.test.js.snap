// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en PrimeraApp Debe de mostrar PrimeraApp correctamente 1`] = `
<Fragment>
  <h1>
    Hola Mundo
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;

exports[`Pruebas en PrimeraApp debe mostrar el subtitulo enviado por props 1`] = `
<Fragment>
  <h1>
    Hola Mundo
  </h1>
  <p>
    Hola Mundo
  </p>
</Fragment>
`;
